Analytics predefined intervals bar

mvkv opened the pull request.

mvkv committed:
Added hashtags chart to analytics,
Improved optional chart's canvas creation

mvkv committed:
Added predefined intervals bar in date picker,
Added first fetched story in user details,

powahftw commented on static/css/analytics.css:
```diff
@@ -110,16 +111,60 @@
+  width: 32px;
+  border: 0.5px solid var(--secondary-color-light);
+  color: white;
+  margin: 0 0;
```
margin: 0;?

powahftw commented on static/css/analytics.css:
```diff
@@ -110,16 +111,60 @@
+
+.bar-button {
+  height: 100%;
+  padding: 4px 4px;
```
padding: 4px; ?

powahftw commented on static/css/analytics.css:
```diff
@@ -110,16 +111,60 @@
+  height: 32px;
+  border: none;
+  padding: 0;
+  padding-left: 4px;
```
padding: 0 4px 0 0?

powahftw commented on static/css/analytics.css:
```diff
@@ -110,16 +111,60 @@
+  padding: 0;
+  padding-left: 4px;
+  margin: 0 8px;
+  margin-top: 4px;
```
margin: 4px 8px 0; ?

powahftw commented on static/js/analytics.js:
```diff
@@ -70,17 +78,24 @@ const composeStatisticsFromSingleUserJson = (json) => {
+  const firstFetchedStoryTime = new Date(firstStory["taken_at"] * 1000)
+    .toISOString()
+    .slice(0, 19)
+    .replace("T", " ");
```
You have the same function above, put them in a util function? ISOStringToDate() or whatever you want

powahftw commented on static/js/analytics.js:
```diff
@@ -335,7 +381,28 @@ const addDateFilterEventListener = () => {
+  }
+  document.querySelectorAll(".predefined-interval-btn").forEach((item) => {
+    item.addEventListener('click', () => {
+      const id = item.id
```
Not much value in this variable, just inline it below

powahftw commented on static/js/analytics.js:
```diff
@@ -335,7 +381,28 @@ const addDateFilterEventListener = () => {
+    "3M": 3,
+    "1M": 1, 
+  }
+  document.querySelectorAll(".predefined-interval-btn").forEach((item) => {
```
it's not an item, it's a date-range-selector-button 

Pick a more specific name so it's clearer. (Not necessarily the long one I mentioned above)

powahftw commented on static/js/analytics.js:
```diff
@@ -335,7 +381,28 @@ const addDateFilterEventListener = () => {
 window.onload = async () => {
   await getAndRenderAnalytics();
   addDateFilterEventListener();
+  predefinedIntervalsEventListener();
```
nit: rename to setPredefinedIntervalsEventListener

mvkv committed:
Improved variable names,
added util function to avoid code duplication,
improved css margins and paddings

powahftw committed:
Merge branch 'master' into Analytics-predefined-intervals-bar
